openedx/core/djangoapps/notifications/models.py

```diff
@@ -62,3 +78,35 @@ def get_content_context(self):
 
     def get_user(self):
         return self.user
+
+
+class NotificationPreference(TimeStampedModel):
+    """
+    Model to store notification preferences for users
+
+    .. no_pii:
+    """
+    user = models.ForeignKey(User, related_name="notification_preferences", on_delete=models.CASCADE)
+    course_id = CourseKeyField(max_length=255, blank=True, default=None)
+    notification_preference_config = models.JSONField(default=NOTIFICATION_PREFERENCE_CONFIG)
```

Comment:
I believe this introduced the following warning at LMS startup:

```
notifications.NotificationPreference.notification_preference_config: (fields.E010) JSONField default should be a callable instead of an instance so that it's not shared between all field instances.
	HINT: Use a callable instead, e.g., use `dict` instead of `{}`.
```

Reply:
Thanks @ormsbee for pointing this out. 
I will resolve this in a followup PR.